internal/proxy/proxy.go

```diff
@@ -59,6 +59,10 @@ type Proxy interface {
 
 	// ObjectExists indicates whether or not any version of the provided object is already present in the proxy.
 	ObjectExists(obj client.Object) (bool, error)
+
+	// IsReady returns true if the proxy is considered ready.
+	// A ready proxy has configuration available and can handle traffic.
+	IsReady() bool
```

Comment:
Do you already decide to make this Readiness Check interface a little bit more meaningful ?

Reply:
I'm not sure I understand the question, can you clarify?

Reply:
Looks like you want to keep the Readiness check signature "IsReady()" as it is. not a blocker.